test/controllers/projects_controller_test.rb

```diff
@@ -55,4 +56,334 @@ class ProjectsControllerTest < ActionDispatch::IntegrationTest
 
     assert_redirected_to projects_url
   end
+
+  # Tests for validate_metadata_csv
```

Comment:
Could consider extracting into a dedicated file as it's getting pretty long?

Reply:
Done.